projects/igniteui-angular/src/lib/grid/row.component.ts

```diff
@@ -30,72 +30,151 @@ import { IgxGridComponent, IRowSelectionEventArgs } from './grid.component';
+     *
+     * ```typescript
+     * // get the row data for the first selected row
+     * let selectedRowData = this.grid.selectedRows[0].rowData;
+     * ```
+     */
     @Input()
     public rowData: any;
 
+    /**
+     * The index of the row.
+     *
+     * ```typescript
+     * // get the index of the second selected row
+     * let selectedRowIndex = this.grid.selectedRows[1].index;
+     * ```
+     */
     @Input()
     public index: number;
 
+    /**
+     * @hidden
+     */
     @Input()
     public gridID: string;
 
+    /**
+     * @hidden
+     */
     @ViewChild('igxDirRef', { read: IgxForOfDirective })
     public virtDirRow: IgxForOfDirective<any>;
 
+    /**
+     * @hidden
+     */
     @ViewChild(forwardRef(() => IgxCheckboxComponent), {read: IgxCheckboxComponent})
     public checkboxElement: IgxCheckboxComponent;
 
+    /**
+     * The rendered cells in the row component.
+     *
+     * ```typescript
+     * // get the cells of the third selected row
+     * let selectedRowCells = this.grid.selectedRows[2].cells;
+     * ```
+     */
     @ViewChildren(forwardRef(() => IgxGridCellComponent), { read: IgxGridCellComponent })
     public cells: QueryList<IgxGridCellComponent>;
 
+    /**
+     * @hidden
+     */
     @HostBinding('style.height.px')
     get rowHeight() {
         return this.grid.rowHeight;
     }
+
+    /**
+     * @hidden
+     */
     @HostBinding('attr.tabindex')
     public tabindex = 0;
 
+    /**
+     * @hidden
+     */
     @HostBinding('attr.role')
     public role = 'row';
 
+    /**
+     * @hidden
+     */
     @HostBinding('class')
     get styleClasses(): string {
         const indexClass = this.index % 2 ? this.grid.evenRowCSS : this.grid.oddRowCSS;
         const selectedClass = this.isSelected ? 'igx-grid__tr--selected' : '';
         return `${this.defaultCssClass} ${indexClass} ${selectedClass}`;
     }
 
+
+    /**
+     * @hidden
+     */
     get focused(): boolean {
         return this.isFocused;
     }
 
+    /**
+     * @hidden
+     */
     set focused(val: boolean) {
         this.isFocused = val;
     }
 
+    /**
+     * @hidden
+     */
     get columns(): IgxColumnComponent[] {
         return this.grid.visibleColumns;
     }
 
+    /**
+     * @hidden
+     */
     get pinnedColumns(): IgxColumnComponent[] {
         return this.grid.pinnedColumns;
     }
 
+    /**
+     * @hidden
+     */
     get unpinnedColumns(): IgxColumnComponent[] {
         return this.grid.unpinnedColumns;
     }
 
+    /**
+     * @hidden
+     */
     public get rowSelectable() {
         return this.grid.rowSelectable;
     }
 
+    /**
+     * @hidden
+     */
     @HostBinding('attr.aria-selected')
     public isSelected: boolean;
 
+    /**
+     * A reference to the grid containing the row.
+     *
+     * ```typescript
+     * // get a reference to the grid of the first selected row
+     * let selectedRowGrid = this.grid.selectedRows[0].grid;
+     * ```
```

Comment:
This snippet is OK, but it will make more sense, when used in the context of a row:

<igx-grid #grid1 [data]="data" [primaryKey]="'ProductID'"
            [rowSelectable]="selection" (onRowSelectionChange)="handleRowSelection($event)" >
</igx-grid>

private onRowSelectionChange(event) {
        const grid = event.row.grid;
    }